src/ol/interaction/extentinteraction.js

```diff
@@ -0,0 +1,429 @@
+  };
+  this.createOrUpdateExtentFeature = function(extent) {
+    var bboxFeature = this.bboxFeature_;
+
+    if (!extent) {
+      extent = [0,0,0,0];
+    }
+    if (!bboxFeature) {
+      bboxFeature = new ol.Feature(new ol.geom.Polygon([[
+        [extent[0], extent[1]],
+        [extent[2], extent[1]],
+        [extent[2], extent[3]],
+        [extent[0], extent[3]],
+        [extent[0], extent[1]]
+      ]], ol.geom.GeometryLayout.XY));
+
+      this.bboxFeature_ = bboxFeature;
+      this.bboxOverlay_.getSource().addFeature(bboxFeature);
+    } else {
+      var geometry = (bboxFeature.getGeometry());
+      geometry.setCoordinates([[
+        [extent[0], extent[1]],
+        [extent[2], extent[1]],
+        [extent[2], extent[3]],
+        [extent[0], extent[3]],
+        [extent[0], extent[1]]
+      ]]);
+    }
+    return bboxFeature;
+  };
+  this.removeExtentFeature = function() {
+    var bboxFeature = this.bboxFeature_;
+    if (bboxFeature) {
+      this.bboxOverlay_.getSource().removeFeature(bboxFeature);
+      this.bboxFeature_ = null;
+    }
+  };
+  this.createOrUpdatePointerFeature = function(vertex) {
+    var vertexFeature = this.vertexFeature_;
+    if (!vertexFeature) {
+      vertexFeature = new ol.Feature(new ol.geom.Point(vertex));
+      this.vertexFeature_ = vertexFeature;
+      this.vertexOverlay_.getSource().addFeature(vertexFeature);
+    } else {
+      var geometry = (vertexFeature.getGeometry());
+      geometry.setCoordinates(vertex);
+    }
+    return vertexFeature;
+  };
+  this.removePointerFeature = function() {
+    var vertexFeature = this.vertexFeature_;
+    if (vertexFeature) {
+      this.vertexOverlay_.getSource().removeFeature(vertexFeature);
+      this.vertexFeature_ = null;
+    }
+  };
+};
+goog.inherits(ol.interaction.Extent, ol.interaction.Pointer);
+
+/* Public Methods */
+/**
+ * @inheritDoc
+ */
+ol.interaction.Extent.prototype.setMap = function(map) {
+  this.bboxOverlay_.setMap(map);
+  this.vertexOverlay_.setMap(map);
+  ol.interaction.Interaction.prototype.setMap.call(this, map);
+};
+/**
+ * Returns the current drawn extent in the requested projection
+ *
+ * @param {ol.proj.Projection=} projection Projection to retun the extent in
+ * @return {ol.Extent} Drawn extent in the passed projection, or in the native projection if undefined.
+ */
+ol.interaction.Extent.prototype.getExtent = function(projection) {
+  var srcProjection = this.getMap().getView().getProjection();
+  var extent = this.extent;
+  if (extent && projection && projection != srcProjection) {
+    extent = ol.proj.transformExtent(extent, srcProjection, projection);
+  }
+  return extent;
+};
+/**
+ * Manually sets the drawn extent, in the requested projection. Uses the native projection if undefined.
+ *
+ * @param {ol.Extent} extent Extent
+ * @param {ol.proj.Projection|string=} projection Projection to return the extent in
+ */
+ol.interaction.Extent.prototype.setExtent = function(extent, projection) {
```

Comment:
If you decide to make this `@api`, I'd go without the `projection` argument here and the `projection` constructor option, and use the view projection. Otherwise it will be confusing for users, because we do not do such thing elsewhere.